inkscape: 1.2.2 -> 1.3, lib2geom: 1.2.2 -> 1.3

leiserfg opened the pull request:
- inkscape: 1.2.2 -> 1.3, lib2geom: 1.2.2 -> 1.3

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

leiserfg committed:
inkscape: 1.2.2 → 1.3

https://inkscape.org/doc/release_notes/1.3/Inkscape_1.3.html

libepoxy for experimental GPU-accelerated rendering.
pyparsing for HPGL support.

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

leiserfg committed:
lib2geom: 1.2.2 → 1.3

https://gitlab.com/inkscape/lib2geom/-/releases/1.3

Co-authored-by: Jan Tojnar <jtojnar@gmail.com>

leiserfg commented:
I updated both in a single commit because Inkscape depends on lib2geom (same authors) and is the only dependant of it.

jtojnar commented:
Unfortunately this breaks on Darwin https://gitlab.com/inkscape/lib2geom/-/issues/62 and aarch64-linux https://gitlab.com/inkscape/lib2geom/-/issues/63

leiserfg commented:
Then we will have to wait for a `1.3.1`

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/update-inkscape-1-3/31570/2

MultisampledNight commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -58,11 +59,11 @@ let
+  version = "1.3";
 
   src = fetchurl {
     url = "https://media.inkscape.org/dl/resources/file/inkscape-${version}.tar.xz";
```
The release notes at https://gitlab.com/inkscape/inkscape/-/releases/INKSCAPE_1_3 suggest to switch to this scheme instead:

```suggestion
    url = "https://inkscape.org/release/inkscape-${version}/source/archive/xz/dl/inkscape-${version}.tar.xz";
```

leiserfg replied:
I will do the fix when the other two issues upstream get fixed.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/inkscape-possible-domain-compromise/32008/5

jtojnar commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -38,6 +38,7 @@
+, libepoxy
 , zlib
 }:
 let
```
Looking at the [list of required Python modules](https://inkscape.gitlab.io/extensions/python-dependencies.txt) linked from changelog, we also need the following:

- pyparsing (used by newly added hpgl parser extension)

The following modules listed do not appear to be used by anything:

- tinycss2
- filelock

jtojnar replied:
I tried running `docs/poetry.sh` in `share/extensions` directory but could not reproduce the list – I got the following on both `INKSCAPE_1_3` and `master` bracnches:

```markdown
# Packages Used Directly

You must always include these in your package:

 * cssselect 1.2.0
 * lxml 4.9.3
 * numpy 1.25.2
 * packaging 23.1
 * pillow 9.5.0
 * pygobject 3.44.1
 * pyparsing 3.1.1
 * pyserial 3.5
 * scour 0.37


# Packages Used Indirectly

You may or may not need to include these, depending on if your packaging system automatically pulls in sub-deps:

 * pycairo >=1.16,<2.0
 * six >=1.9.0
```

jtojnar commented on pkgs/development/libraries/lib2geom/default.nix:
```diff
@@ -15,27 +15,17 @@
-      sha256 = "RMgwJkylrGFTTrqBzqs5j2LMSLsHhcE/UT1pKBZnU50=";
-    })
-  ];
-
```
Please cherry-pick https://gitlab.com/inkscape/lib2geom/-/commit/23d9393af4bee17aeb66a3c13bdad5dbed982d08.patch (might need other commits from https://gitlab.com/inkscape/lib2geom/-/merge_requests/102 to apply) to get it to build with clang (on Darwin).

jtojnar replied:
Per [upstream recommendation](https://gitlab.com/inkscape/lib2geom/-/issues/63#note_1526186398), please add something like the following under the `doCheck` to make it build on aarch64:

```nix
  # TODO: Update cmake hook to make it simpler to selectively disable cmake tests: #113829
  checkPhase = let
    disabledTests =
      lib.optionals stdenv.isAarch64 [
        # Broken on all platforms, test just accidentally passes on some.
        # https://gitlab.com/inkscape/lib2geom/-/issues/63
        "elliptical-arc-test"
      ];
  in ''
    runHook preCheck
    ctest --output-on-failure -E '^${lib.concatStringsSep "|" disabledTests}$'
    runHook postCheck
  '';
```

jtojnar commented:
> I updated both in a single commit because Inkscape depends on lib2geom (same authors) and is the only dependant of it.

Separate commits are generally preferred. Then you can link the relevant changelogs in the commit message easily, among other benefits. If you update lib2geom first, Inkscape should still work, unless the library broke the API.

TheLostLambda commented:
Is this still waiting on upstream fixes, or is this close to merging?

Let me know if I can help any!

leiserfg commented:
> Is this still waiting on upstream fixes, or is this close to merging?

Check
https://gitlab.com/inkscape/lib2geom/-/issues/63

leiserfg commented:
@jtojnar  can you take over this? I can't test it on Arm anyways.

jtojnar committed:
python3.pkgs.inkex: inherit source from Inkscape

This partly reverts 5d12e405f6ed8f061673f5ce609b5e74fb135f34 to avoid potential conflicts.
Also move dependencies only introduced because of tests to checkInputs.

jtojnar reviewed (approved):
I tested some basic operations and the hpgl import. Console complains about resource missing but apply-transform extension still appears to work and either way, it is not regression compared to 1.2:

```ShellSession

$ nix-shell -I nixpkgs=$PWD -p "inkscape-with-extensions.override { inkscapeExtensions = null; }" --run inkscape
** (org.inkscape.Inkscape:252608): WARNING **: 22:38:53.378: Failed to find resource file 'inkex.py'. Looked in:
	/home/jtojnar/.config/inkscape/extensions/inkex.py
	(null)
	/nix/store/cyfbhp0rx19gcrjmdka3vryy0bhnhl09-inkscape-with-extensions-1.3/share/inkscape/extensions/inkex.py

** (org.inkscape.Inkscape:252608): WARNING **: 22:38:53.786: You have a huge number of font families (1711), and Cairo is limiting the size of widgets you can draw.
Your preview cell height is capped to 17.
```

raboof reviewed (approved):
From https://gitlab.com/inkscape/lib2geom/-/issues/63 I get the impression that we shouldn't let that issue hold up the merge, right?

Tested this build with `nix-shell -p "inkscape-with-extensions.override { inkscapeExtensions = [ inkscape-extensions.inkcut ]; }" --run inkscape`, creating a simple image and opening it in inkcut. Unfortunately there seems to be something broken with the way inkcut converts objects to paths, but that seems to be an inkscape/inkcut problem, not a reason to hold back this PR IMHO (as inkcut nixpkgs package maintainer).

jtojnar commented:
Status of filelock dependency still unclear, see https://github.com/NixOS/nixpkgs/pull/258885#issuecomment-1747607537

arkivm commented:
v1.3 is still broken on darwin, and v1.3.1 is not going to be a [long wait](https://gitlab.com/groups/inkscape/-/milestones/14#tab-issues) (Nov 15).

jtojnar committed:
inkscape: inherit filelock dependency from cachecontrol

The dependency is not imported anywhere in the Inkscape code so let’s make the requirement explicit.

wegank committed:
inkscape: fix runtime error on darwin

wegank commented:
@ofborg build inkscape inkscape.passthru.tests

raboof commented:
> Tested this build with `nix-shell -p "inkscape-with-extensions.override { inkscapeExtensions = [ inkscape-extensions.inkcut ]; }" --run inkscape`, creating a simple image and opening it in inkcut. Unfortunately there seems to be something broken with the way inkcut converts objects to paths, but that seems to be an inkscape/inkcut problem

(created https://github.com/inkcut/inkcut/issues/374 for this problem)

raboof commented:
> > Tested this build with `nix-shell -p "inkscape-with-extensions.override { inkscapeExtensions = [ inkscape-extensions.inkcut ]; }" --run inkscape`, creating a simple image and opening it in inkcut. Unfortunately there seems to be something broken with the way inkcut converts objects to paths, but that seems to be an inkscape/inkcut problem
> 
> (created [inkcut/inkcut#374](https://github.com/inkcut/inkcut/issues/374) for this problem)

(not sure what changed, but this works fine now)
